packages/jupyter-ai/src/completions/plugin.ts

```diff
@@ -68,79 +88,100 @@ export const inlineCompletionProvider: JupyterFrontEndPlugin<void> = {
+     */
+    async function updateJaiIcpSettings(
+      newJaiIcpSettings: Partial<JaiInlineProvider.ISettings>
+    ) {
+      const icSettings = await getIcSettings();
+      const oldUserIcpSettings = icSettings.user.providers;
+      const newUserIcpSettings = {
+        ...oldUserIcpSettings,
+        [JaiInlineProvider.ID]: {
+          ...oldUserIcpSettings?.[JaiInlineProvider.ID],
+          ...newJaiIcpSettings
         }
-        const providers = Object.assign({}, settings.user.providers) as any;
-        const ourSettings = {
-          ...JupyterAIInlineProvider.DEFAULT_SETTINGS,
-          ...providers[provider.identifier]
-        };
-        const wasEnabled = ourSettings['enabled'];
-        providers[provider.identifier]['enabled'] = !wasEnabled;
-        settings.set('providers', providers);
+      };
+      icSettings.set('providers', newUserIcpSettings);
+    }
+
+    app.commands.addCommand(CommandIDs.toggleCompletions, {
+      execute: async () => {
+        const jaiIcpSettings = await getJaiIcpSettings();
+        updateJaiIcpSettings({
+          enabled: !jaiIcpSettings.enabled
+        });
       },
-      label: 'Enable Jupyternaut Completions',
+      label: 'Enable completions by Jupyternaut',
       isToggled: () => {
         return provider.isEnabled();
       }
     });
 
     app.commands.addCommand(CommandIDs.toggleLanguageCompletions, {
-      execute: () => {
+      execute: async () => {
+        const jaiIcpSettings = await getJaiIcpSettings();
         const language = findCurrentLanguage();
-        if (!settings || !language) {
+        if (!language) {
           return;
         }
-        const providers = Object.assign({}, settings.user.providers) as any;
-        const ourSettings = {
-          ...JupyterAIInlineProvider.DEFAULT_SETTINGS,
-          ...providers[provider.identifier]
-        };
-        const wasDisabled = ourSettings['disabledLanguages'].includes(
-          language.name
-        );
-        const disabledList: string[] =
-          providers[provider.identifier]['disabledLanguages'];
-        if (wasDisabled) {
-          disabledList.filter(name => name !== language.name);
-        } else {
-          disabledList.push(language.name);
-        }
-        settings.set('providers', providers);
+
+        const disabledLanguages = [...jaiIcpSettings.disabledLanguages];
+        const newDisabledLanguages = disabledLanguages.includes(language.name)
+          ? disabledLanguages.filter(l => l !== language.name)
+          : disabledLanguages.concat(language.name);
+
+        updateJaiIcpSettings({
+          disabledLanguages: newDisabledLanguages
+        });
       },
       label: () => {
         const language = findCurrentLanguage();
         return language
-          ? `Enable Completions in ${displayName(language)}`
-          : 'Enable Completions for Language of Current Editor';
+          ? `Disable completions for ${displayName(language)} files`
```

Comment:
Using "files" word here may be confusing for the user. Is a cell in a notebook a file? Is a console cell a file? Etc.
"Document" is a term we use to encompass both notebooks and files but it would sound odd to me too. Maybe "code"? But then, "completions for Markdown code" or "text code" sounds odd.

Reply:
I agree that `files` is unnecessary. The old style of `Enable completions in {language}` is more concise and still understandable.

Reply:
The latest revision will have `Disable completions in {language}`. I prefer "Disable..." over "Enable..." here since this way, the menu item can be un-checked by default. I think that makes more sense.